internal/zinc-classfile/src/main/scala/sbt/internal/inc/classfile/Analyze.scala

```diff
@@ -55,9 +52,9 @@ private[sbt] object Analyze {
         .orElse(Option(loadedClass.getEnclosingClass).flatMap(binaryToSourceName))
 
       srcClassName match {
-        case Some(srcClassName) =>
-          analysis.generatedNonLocalClass(source, newClass, binaryClassName, srcClassName)
-          classNames += srcClassName
+        case Some(className) =>
+          analysis.generatedNonLocalClass(source, newClass, binaryClassName, className)
+          classNames += className
```

Comment:
Good observation, this was warning generated by intellij "Suspicious shadowing by a Variable Pattern". I don't have strong opinion how to approach this but usually I try to avoid shadowing by using different names